managed/models/database.go

```diff
@@ -1149,79 +1168,103 @@ func SetupDB(ctx context.Context, sqlDB *sql.DB, params SetupDBParams) (*reform.
 		return nil, errCV
 	}
 
-	agentColumnsToEncrypt := []encryption.Column{
-		{Name: "username"},
-		{Name: "password"},
-		{Name: "aws_access_key"},
-		{Name: "aws_secret_key"},
-		{Name: "mongo_db_tls_options", CustomHandler: EncryptMongoDBOptionsHandler},
-		{Name: "azure_options", CustomHandler: EncryptAzureOptionsHandler},
-		{Name: "mysql_options", CustomHandler: EncryptMySQLOptionsHandler},
-		{Name: "postgresql_options", CustomHandler: EncryptPostgreSQLOptionsHandler},
-		{Name: "agent_password"},
-	}
-
-	itemsToEncrypt := []encryption.Table{
-		{
-			Name:        "agents",
-			Identifiers: []string{"agent_id"},
-			Columns:     agentColumnsToEncrypt,
-		},
-	}
-
-	if err := migrateDB(db, params, itemsToEncrypt); err != nil {
+	if err := migrateDB(db, params, DefaultAgentEncryptionColumns); err != nil {
 		return nil, err
 	}
 
 	return db, nil
 }
 
 // EncryptDB encrypts a set of columns in a specific database and table.
-func EncryptDB(tx *reform.TX, params SetupDBParams, itemsToEncrypt []encryption.Table) error {
-	if len(itemsToEncrypt) == 0 {
+func EncryptDB(tx *reform.TX, database string, itemsToEncrypt []encryption.Table) error {
+	return dbEncryption(tx, database, itemsToEncrypt, encryption.EncryptItems, encryptionExists, addToEncryptedItems)
+}
+
+func encryptionExists(m map[string]bool, key string) bool {
+	return m[key]
+}
+
+func addToEncryptedItems(encryptedItems []string, items []string) []string {
+	return slices.Concat(encryptedItems, items)
+}
+
+// DecryptDB decrypts a set of columns in a specific database and table.
+func DecryptDB(tx *reform.TX, database string, itemsToEncrypt []encryption.Table) error {
+	return dbEncryption(tx, database, itemsToEncrypt, encryption.DecryptItems, encryptionNotExists, removeFromEncryptedItems)
+}
+
+func encryptionNotExists(m map[string]bool, key string) bool {
+	return !encryptionExists(m, key)
+}
+
+func removeFromEncryptedItems(encryptedItems []string, items []string) []string {
+	res := []string{}
+	for _, encryptedItem := range encryptedItems {
+		exists := false
+		for _, item := range items {
+			if encryptedItem == item {
+				exists = true
+			}
+		}
+
+		if exists {
+			continue
+		}
+
+		res = append(res, encryptedItem)
+	}
+
+	return res
+}
+
+func dbEncryption(tx *reform.TX, database string, items []encryption.Table,
+	encryptionHandler func(tx *reform.TX, tables []encryption.Table) error,
+	checkHandler func(m map[string]bool, key string) bool,
```

Comment:
checkHandler could be replaced with `expectedState` boolean and comparing `m[key] == expectedState`